tutorials/integrate_external_libs_into_pipeline/README.md

```diff
@@ -1,31 +1,251 @@
+
+Please refer to the [MatX documentation](https://nvidia.github.io/MatX/build.html) for detailed instructions on building and using the MatX library.
+
+The following is a sample CMakeLists.txt file for a project that uses MatX:
+
+```
+cmake_minimum_required(VERSION 3.20)
+project(my_app CXX)
+
+# Holoscan
+find_package(holoscan 2.0 REQUIRED CONFIG
+             PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
+
+# Enable cuda language
+set(CMAKE_CUDA_ARCHITECTURES "70;80")
+enable_language(CUDA)
+
+# Uncomment below for debug build
+# set(CMAKE_BUILD_TYPE Debug)
+
+# Download MatX
+# OBS: Currently pulls a specific commit of MatX (with host support for find_idx)
+include(FetchContent)
+FetchContent_Declare(
+  MatX
+  GIT_REPOSITORY https://github.com/NVIDIA/MatX.git
+  GIT_TAG main
+)
+FetchContent_MakeAvailable(MatX)
 
-In this tutorial, we will show how to integrate the libraries below into Holoscan applications in Python:
-- [Integrate RAPIDS **cuCIM** library](#integrate-rapids-cucim-library)
-- [Integrate RAPIDS **CV-CUDA** library](#integrate-cv-cuda-library)
-- [Integrate **OpenCV with CUDA Module**](#integrate-opencv-with-cuda-module)
+add_executable(my_app
+  my_app.cpp
+)
 
+target_link_libraries(my_app
+  PRIVATE
+  holoscan::core
+  holoscan::ops::aja
+  holoscan::ops::video_stream_replayer
+  holoscan::ops::format_converter
+  holoscan::ops::inference
+  holoscan::ops::segmentation_postprocessor
+  holoscan::ops::holoviz
+  matx::matx
+)
+
+```
+
+#### Sample code
+
+The following are the sample applications that use MatX library to integrate with Holoscan SDK.
+
+- [Multi AI Application with SSD Detection and MONAI Endoscopic Tool Segmentation](https://github.com/nvidia-holoscan/holohub/tree/main/applications/multiai_endoscopy)
+  - `applications/multiai_endoscopy`
+- [Network Radar Pipeline](https://github.com/nvidia-holoscan/holohub/tree/main/applications/network_radar_pipeline/cpp)
+  - `applications/network_radar_pipeline/cpp`
+- [Simple Radar Pipeline Application](https://github.com/nvidia-holoscan/holohub/tree/main/applications/simple_radar_pipeline/cpp)
+  - `applications/simple_radar_pipeline/cpp`
+
+**On the GPU**
+
+- <https://github.com/nvidia-holoscan/holohub/blob/main/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/multi_ai.cu>
+
+```cpp
+#include <holoscan/holoscan.hpp>
+#include <matx.h>
+
+// ...
+
+void compute(InputContext& op_input, OutputContext& op_output,
+             ExecutionContext& context) override {
+  // Get input message and make output message
+  auto in_message = op_input.receive<gxf::Entity>("in").value();
+  // ...
+  auto boxes = in_message.get<Tensor>("inference_output_detection_boxes");
+  auto scores = in_message.get<Tensor>("inference_output_detection_scores");
+  int32_t Nb = scores->shape()[1];  // Number of boxes
+  // ...
+  auto boxesl_mx = matx::make_tensor<float>({1, Nl(), 4});
```

Comment:
This example is an excerpt of [this code](https://github.com/nvidia-holoscan/holohub/blob/main/applications/multiai_endoscopy/cpp/post-proc-matx-gpu/multi_ai.cu#L107) to show users how using MatX looks in the code. I think showing the full code is not necessary.

Let me update it as shown below to avoid confusion with `Nl`:

```
  int32_t Nb = scores->shape()[1];  // Number of boxes
  auto Nl = matx::make_tensor<int>({});  // Number of label boxes
  // ...
```


Let me